liota/lib/transports/amqp.py

```diff
@@ -0,0 +1,343 @@
+            else:
+                client_cert_available = False
+
+            # Validate client key file path
+            if self.edge_system_identity.key_file:
+                if os.path.exists(self.edge_system_identity.key_file):
+                    client_key_available = True
+                else:
+                    log.error("Error : Wrong client key path.")
+                    raise ValueError("Error : Wrong client key path.")
+            else:
+                client_key_available = False
+
+            """
+            Certificate Validations
+            #1. If client certificate is not present throw error
+            #2. If client key is not present throw error
+
+            If both client certificate and keys are not available, proceed with root CA
+            """
+
+            if not client_cert_available and client_key_available:
+                log.error("Error : Client key found, but client certificate not found")
+                raise ValueError("Error : Client key found, but client certificate not found")
+            if client_cert_available and not client_key_available:
+                log.error("Error : Client key found, but client certificate not found")
+                raise ValueError("Error : Client certificate found, but client key not found")
+
+            """
+            Setup ssl options
+            """
+            ssl_details = ({"ca_certs": self.remote_system_identity.root_ca_cert,
+                            "certfile": self.edge_system_identity.cert_file,
+                            "keyfile": self.edge_system_identity.key_file,
+                            "cert_reqs": getattr(ssl, self.tls_details.cert_required),
+                            "ssl_version": getattr(ssl, self.tls_details.tls_version),
+                            "ciphers": self.tls_details.cipher})
+
+            try:
+                """
+                Establish connection with
+                a) Certificate based authorization
+                b) Certificate based authorization and username/password based authentication
+                """
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=self.host, port=self.port,
+                                                                                    credentials=credentials,
+                                                                                    ssl=True, ssl_options=ssl_details))
+                self.channel = self.connection.channel()
+            except Exception, e:
+                log.exception("Unable to establish connection")
+                raise e
+
+        else:
+            try:
+                """
+                Establish connection with
+                a) Plain AMQP
+                b) Username/password based authentication
+                """
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=self.host, port=self.port,
```

Comment:
Previous connection creation and this block can be kept together so that it looks better.  This also also helps us in handling connect disconnect timeout mechanism